h/accounts/views.py

```diff
@@ -234,126 +240,98 @@ class AsyncRegisterController(RegisterController):
     __view_mapper__ = AsyncFormViewMapper
 
 
-def _emails_must_match_validator(form, value):
-    """Raise colander.Invalid if "email" and "emailAgain" don't match."""
-    if value.get("email") != value.get("emailAgain"):
-        exc = colander.Invalid(form, "The emails must match")
-        exc["emailAgain"] = "The emails must match."
-        raise exc
-
-
-class _InvalidEditProfileRequestError(Exception):
-
-    """Raised if validating an edit user profile request fails."""
-
-    def __init__(self, errors):
-        super(_InvalidEditProfileRequestError, self).__init__()
-        self.errors = errors
-
-
-def _validate_edit_profile_request(request):
-    """Validate the given request using the EditProfileSchema.
-
-    :returns: if the request is valid returns a Deform "appstruct" with keys
-        ``"username"``, ``"pwd"`` and ``"email"``
-    :rtype: dict
-
-    :raises _InvalidEditProfileRequestError: if the request is invalid
-
-    """
-    schema = schemas.EditProfileSchema(
-        validator=_emails_must_match_validator).bind(request=request)
-    form = deform.Form(schema)
-    try:
-        return form.validate(request.POST.items())
-    except deform.ValidationFailure as err:
-        raise _InvalidEditProfileRequestError(errors=err.error.children)
-
-
 @view_auth_defaults
 @view_config(attr='edit_profile', route_name='edit_profile')
 @view_config(attr='disable_user', route_name='disable_user')
 @view_config(attr='profile', route_name='profile')
-class ProfileController(horus.views.ProfileController):
+class ProfileController(object):
+    def __init__(self, request):
+        self.request = request
+        self.schema = schemas.ProfileSchema().bind(request=self.request)
+        self.form = deform.Form(self.schema)
+
     def edit_profile(self):
-        try:
-            appstruct = _validate_edit_profile_request(self.request)
-        except _InvalidEditProfileRequestError as err:
-            return dict(errors=err.errors)
+        """Handle POST payload from profile update form."""
+        if self.request.method != 'POST':
```

Comment:
This can also be done by adding the `request_method` predicate to the view decorator, but this is okay.